### ISSUE TITLE
I can do training, but somehow the validation inside the training not work

### ISSUE THREAD

**Issue by saikanov** [opendatalab/DocLayout-YOLO#184]
<img width="783" height="101" alt="Image" src="https://github.com/user-attachments/assets/353ed1af-912b-477c-983e-257099f5dc62" />

i got his from csv file in the trainig result

and usually there is a delay between epoch to validation, so validation doesnt occur in my case

could someone please help me? i cant tell if my model good enough or not.

**Comment by saikanov** [comment]
<img width="1326" height="669" alt="Image" src="https://github.com/user-attachments/assets/abe07155-b98c-439c-9d00-79423b8088cf" />

Update: the validation is running, but the results.csv val cls etc still zero

<img width="1490" height="348" alt="Image" src="https://github.com/user-attachments/assets/9ea8a9d1-a2a7-4564-88e3-6245c5fc81b9" />

is this a normal thing?

**Comment by JulioZhao97** [comment]
@saikanov Hello! This is set by design, you can change code here (lines I comment out)

https://github.com/opendatalab/DocLayout-YOLO/blob/32a8ec276b3d79bf40561c4bc4b8e21ef32ac6fd/doclayout_yolo/engine/validator.py#L172-L208

**Comment by saikanov** [comment]
> [@saikanov](https://github.com/saikanov) Hello! This is set by design, you can change code here (lines I comment out)
> 
> [DocLayout-YOLO/doclayout_yolo/engine/validator.py](https://github.com/opendatalab/DocLayout-YOLO/blob/32a8ec276b3d79bf40561c4bc4b8e21ef32ac6fd/doclayout_yolo/engine/validator.py#L172-L208)
> 
> Lines 172 to 208 in [32a8ec2](/opendatalab/DocLayout-YOLO/commit/32a8ec276b3d79bf40561c4bc4b8e21ef32ac6fd)
> 
>  # HACK: don't record loss!!! 
>  self.run_callbacks("on_val_start") 
>  dt = ( 
>      Profile(device=self.device), 
>      # Profile(device=self.device), 
>      Profile(device=self.device), 
>      Profile(device=self.device), 
>  ) 
>  bar = TQDM(self.dataloader, desc=self.get_desc(), total=len(self.dataloader)) 
>  self.init_metrics(de_parallel(model)) 
>  self.jdict = []  # empty before each val 
>  for batch_i, batch in enumerate(bar): 
>      self.run_callbacks("on_val_batch_start") 
>      self.batch_i = batch_i 
>      # Preprocess 
>      with dt[0]: 
>          batch = self.preprocess(batch) 
>           
>      # Inference 
>      with dt[1]: 
>          preds = model(batch["img"], augment=augment) 
>   
>      # Loss 
>      # with dt[2]: 
>      #     if self.training: 
>      #         self.loss += model.loss(batch, preds)[1] 
>       
>      # Postprocess 
>      with dt[2]: 
>          preds = self.postprocess(preds, conf=self.args.conf) 
>   
>      self.update_metrics(preds, batch) 
>      if self.args.plots and batch_i < 3: 
>          self.plot_val_samples(batch, batch_i) 
>          self.plot_predictions(batch, preds, batch_i) 
>   
>      self.run_callbacks("on_val_batch_end")

alright thanks i will try it
